DataFormats/FTLRecHit/interface/FTLCluster.h

```diff
@@ -0,0 +1,272 @@
+  inline int maxHitCol() const { return minHitCol() + colSpan();} // The max y index.
+  
+  const std::vector<uint8_t> & hitOffset() const { return theHitOffset;}
+  const std::vector<float> & hitENERGY() const { return theHitENERGY;}
+  const std::vector<float> & hitTIME() const { return theHitTIME;}
+  const std::vector<float> & hitTIME_ERROR() const { return theHitTIME_ERROR;}
+    
+  // infinite faster than above...
+  FTLHit hit(int i) const {
+    return FTLHit(minHitRow() + theHitOffset[i*2],
+		 minHitCol() + theHitOffset[i*2+1],
+		  theHitENERGY[i],
+		  theHitTIME[i],
+		  theHitTIME_ERROR[i]
+		 );
+
+  }
+  
+  FTLHit seed() const {
+    return hit(seed_);
+  }
+  
+  int colSpan() const {return theHitColSpan; }
+  
+  int rowSpan() const { return theHitRowSpan; }
+  
+  const DetId& id() const { return theid; }
+  const DetId& detid() const { return id(); }
+
+  bool overflowCol() const { return overflow_(theHitColSpan); }
+  
+  bool overflowRow() const { return overflow_(theHitRowSpan); }
+  
+  bool overflow() const { return  overflowCol() || overflowRow(); }
+  
+  void packCol(uint16_t ymin, uint16_t yspan) {
+    theMinHitCol = ymin;
+    theHitColSpan = std::min(yspan, uint16_t(MAXSPAN));
+  }
+  void packRow(uint16_t xmin, uint16_t xspan) {
+    theMinHitRow = xmin;
+    theHitRowSpan = std::min(xspan, uint16_t(MAXSPAN));
+  }
+  
+   void setClusterErrorX( float errx ) { err_x = errx; }
+   void setClusterErrorY( float erry ) { err_y = erry; }
+   void setClusterErrorTime( float errtime ) { err_time = errtime; }
+   float getClusterErrorX() const { return err_x; }
+   float getClusterErrorY() const { return err_y; }
+   float getClusterErrorTime() const { return err_time; }
+  
+private:
+
+   DetId theid;
+   
+   std::vector<uint8_t>  theHitOffset;
+   std::vector<float> theHitENERGY;
+   std::vector<float> theHitTIME;
+   std::vector<float> theHitTIME_ERROR;
+   
+   uint16_t theMinHitRow=MAXPOS; // Minimum hit index in the x direction (low edge).
+   uint16_t theMinHitCol=MAXPOS; // Minimum hit index in the y direction (left edge).
+   uint8_t theHitRowSpan=0; // Span hit index in the x direction (low edge).
+   uint8_t theHitColSpan=0; // Span hit index in the y direction (left edge).
+   
+   float err_x=-99999.9f;
+   float err_y=-99999.9f;
+   float err_time=-99999.9f;
+   
+   uint8_t seed_;
+   
+   float weighted_sum(const std::vector<float>& weights, const std::function<float (unsigned int i)>& sumFunc, const std::function<float (float,float)>& outFunc) const 
+   {
+     float tot=0;
+     float sumW=0;
+     for (unsigned int i=0; i<weights.size(); ++i)
+       {
+	 tot += sumFunc(i);
+	 sumW += weights[i];
+       }
+     return outFunc(tot,sumW);
+   }
+   
+   float weighted_mean(const std::vector<float>& weights, const std::function<float (unsigned int)>& value) const
+   {
+     auto sumFunc=[weights,value](unsigned int i) { return weights[i]*value(i); } ;
```

Comment:
```suggestion
     auto sumFunc=[&weights,value](unsigned int i) { return weights[i]*value(i); } ;
```
Capture be reference to avoid copying the entire `std::vector`.